File: Software_Testing_Lab/Triangle.java
import java.util.*;
public class Triangle{
    public static void main(String args[]){
	Scanner sc=new Scanner(System.in);
	System.out.println("Enter values: ");
	while(true){
	    try{	  
		int a=sc.nextInt(),b=sc.nextInt(),c=sc.nextInt();
		while(a<0 || a>10 || b<0 || b>10 || c<0 || c>10){
		    System.out.println("Invalid Input!!Re-enter values: ");
		    a=sc.nextInt();
		    b=sc.nextInt();
		    c=sc.nextInt();
		}
		if(a>=b+c || b>=a+c || c>=a+b)
		    System.out.println("Not a triangle!");
		else if(a==b && b==c)
		    System.out.println("Equilateral triangle");
		else if(a==b || b==c || a==c)
		    System.out.println("Isoscles triangle");
		else
		    System.out.println("Scalene triangle");
		break;
	    }
	    catch(Exception e){
		System.out.println("Invalid Input!!Re-enter values: ");
		sc.nextLine();
		continue;
	    }
	}
	sc.close();
    }
}

File: Software_Testing_Lab/Triangle.py
def isvalid(v): return v>0 and v<=10
a,b,c=-1,-1,-1
while True: 
    try: a,b,c=map(int,input("Enter the sides of the triangle: ").split())
    except: 
        print("Invalid input: Wrong format.")
        continue
    if isvalid(a) and isvalid(b) and isvalid(c): break
    else: print("Invalid input: Number out of bounds.")
if a+b<=c or a+c<=b or b+c<=a: print("Not a triangle!")
elif a==b and b==c: print("Equilateral Triangle")
elif a==b or b==c or a==c: print("Isoscles triangle")
else: print("Scalene Triangle")